translate/v23tov30/v23tov30.go

```diff
@@ -447,3 +447,81 @@ func translateDirectories(dirs []old.Directory, m map[string]string) (ret []type
 	}
 	return
 }
+
+// RemoveDuplicateFilesAndUnits is a helper function that removes duplicated files/units from
+// spec v2 config, since neither spec v3 nor the translator function allow for duplicate file
+// entries in the config.
+// This functionality is not included in the Translate function and has some limitations, but
+// may be useful in cases where configuration has to be sanitized before translation.
+// For duplicates, it takes ordering into consideration by taking the file/unit contents from
+// the slice with the highest index value, which is assumed to be the latest revision.
+// Unit dropins are concat'ed, i.e. if no duplicate dropin of the same name exists it is added
+// to the list of dropins of the deduplicated unit definition.
+// The function will fail if multiple filesystems are defined.
+// It will also fail if file appendices are encountered.
+func RemoveDuplicateFilesAndUnits(cfg old.Config) (old.Config, error) {
+	files := cfg.Storage.Files
+	units := cfg.Systemd.Units
+
+	if len(cfg.Storage.Filesystems) > 1 {
```

Comment:
Just curious, I was under the impression that `root` is always there by default in ignition, and you only had to define "new" filesystems in the filesystems section. Is that not the case?

Reply:
good question. Maybe this should just check on the files and if it's not `root` or `/` it'll fail?

Reply:
Is `Filesystem` ever `/`?

Reply:
There's nothing preventing that from being used afaik, as it's just a label. `root` is definitely used more commonly. I wouldn't mind removing `/` here, please let me know.

Reply:
`root` is the name automatically assigned in the universal base config.  `/` would need to be assigned explicitly in the filesystems section, and I haven't seen that name used before.  Unless we know of an existing user of `/`, I'd suggest removing it.

Reply:
removed